tools/Customization/DevHome.Customization/Views/AddRepositoriesView.xaml.cs

```diff
@@ -37,16 +38,46 @@ public void RemoveFolderButton_Click(object sender, RoutedEventArgs e)
         }
     }
 
-    public void AssignSourceControlProviderButton_Click(object sender, RoutedEventArgs e)
+    public async void AssignSourceControlProviderButton_Click(object sender, RoutedEventArgs e)
     {
         // Extract relevant data from view and give to view model for assign
         MenuFlyoutItem menuItem = (MenuFlyoutItem)sender;
         if (menuItem.DataContext is RepositoryInformation repoInfo)
         {
-            ViewModel.AssignSourceControlProviderToRepository(menuItem.Text, repoInfo.RepositoryRootPath);
+            var taskResult = await ViewModel.AssignSourceControlProviderToRepository(menuItem.Text, repoInfo.RepositoryRootPath);
+            if (taskResult?.Result != Helpers.ResultType.Success)
+            {
+                _log.Error("Error occurred while assigning source control provider: ", taskResult?.Error, taskResult?.Exception, taskResult?.DiagnosticText, taskResult?.DisplayMessage);
+                ShowErrorContentDialog(taskResult!, this.XamlRoot);
+            }
         }
     }
 
+    public async void ShowErrorContentDialog(SourceControlValidationResult result, XamlRoot xamlRoot)
+    {
+        var stringResource = new StringResource("DevHome.Customization.pri", "DevHome.Customization/Resources");
+        string diagnosticText = result.DiagnosticText ?? string.Empty;
+        string content;
+        if (diagnosticText.Contains("Invalid repository path"))
```

Comment:
Switching based on a string search is brittle. Instead, check the ErrorType.

Reply:
The distinct display message from the extension is being used and this string search has been removed.